server/service/osquery_utils/queries.go

```diff
@@ -2058,6 +2059,130 @@ func directIngestMDMDeviceIDWindows(ctx context.Context, logger log.Logger, host
 	return ds.UpdateMDMWindowsEnrollmentsHostUUID(ctx, host.UUID, rows[0]["data"])
 }
 
+var luksVerifyQuery = DetailQuery{
+	Platforms: fleet.HostLinuxOSs,
+	QueryFunc: func(ctx context.Context, logger log.Logger, host *fleet.Host, ds fleet.Datastore) string {
+		emptyQuery := "SELECT 1 WHERE 1 = 0"
+		if !host.IsLUKSSupported() {
+			return emptyQuery
+		}
```

Comment:
Would it be a lot of work to have `QueryFunc` return `(string, bool)` instead of this hack? :)